go-controller/pkg/ovn/pods.go

```diff
@@ -655,26 +655,13 @@ func ovnNBLSPDel(client libovsdbclient.Client, logicalPort, logicalSwitch string
 	return nil
 }
 
-func findLogicalSwitch(nbClient libovsdbclient.Client, logicalSwitchName string) (*nbdb.LogicalSwitch, error) {
-	logicalSwitches := []nbdb.LogicalSwitch{}
-	ctx, cancel := context.WithTimeout(context.Background(), ovntypes.OVSDBTimeout)
-	defer cancel()
-	err := nbClient.WhereCache(
-		func(ls *nbdb.LogicalSwitch) bool {
-			return ls.Name == logicalSwitchName
-		}).List(ctx, &logicalSwitches)
-
-	if err != nil {
-		return nil, fmt.Errorf("error finding logical switch %s: %v", logicalSwitchName, err)
-	}
-
-	if len(logicalSwitches) == 0 {
-		return nil, libovsdbclient.ErrNotFound
-	}
-
-	if len(logicalSwitches) > 1 {
-		return nil, fmt.Errorf("unexpectedly found multiple logical switches: %v", logicalSwitches)
+func (oc *Controller) findLogicalSwitch(nbClient libovsdbclient.Client, logicalSwitchName string) (*nbdb.LogicalSwitch, error) {
```

Comment:
@JacobTanenbaum should we bother pretending to return the full LogicalSwitch here? Probably not, given we don't fill everything in, yet the code that calls this function would previously get the full LS record including all properties. We should probably just return the UUID as a string and then a bool if it's found so we can use the `lsuuid, ok := findLogicalSwitch(name); !ok {` pattern.

Also, be careful because callers of this function are testing `libovsdb.ErrNotFound` but this function will no longer return that error... you'd want to convert callers to the `!ok` pattern then.

Reply:
good call on the returned error code but I returned the switch because the UUID identifies a logicalSwitch and is the only field needed to call for mutations to the switch. @dcbw still think I should change it?

Reply:
@JacobTanenbaum yeah, I think if we're not going to return a fully populated LogicalSwitch struct we should just return the UUID. We should also change the name to `findLogicalSwitchUUID` too.

Also, since you're making findLogicalSwitch[UUID] a method of `Controller` we no longer need to pass the libosdb client in since you can just use `oc.nbClient` directly in the function.

Reply:
Or wait, we don't even use the passed-in nbClient anymore do we... and actually maybe we just remove this function entirely and let users call oc.lsManager.GetUUID() directly?